### ISSUE TITLE
[L04] Approved proposals may be executed at any time

### ISSUE THREAD

**Issue by yorhodes** [celo-org/celo-monorepo#8329]
After a proposal reaches the Approved stage, it can be executed at any time after the &#8220;veto period&#8221; ends. However, note that proposals do not expire. Thus, a proposal can lay dormant after the veto period until it is profitable to perform it.
This issue is of low severity due to a few factors which hinder it&#8217;s probability. The first is that a proposal may also be cancelled at any time after it is approved. Another factor is that anyone can execute a proposal once it is approved.
We mention this issue to make it clear to users that dormant proposals can lead to abuse. In the event that the network is experiencing high congestion, it may seem like a waste to approve or cancel a proposal due to high fees. However, it should be made clear that dormant proposals being executed at rates that differ substantially from current price of CELO can destabilize markets and drain the Celo reserves. Consider informing users, and perhaps establishing a proposal expiry time, after which approved proposals can no longer be executed.

**Comment by tkporter** [comment]
After brief discussion, we think this is worth informing users (both exchangers & community members) this can happen, but won't make any smart contract changes to implement an expiry (so essentially a `wontfix`).

We believe that adding an expiry could leave the exchanger with more opportunity for malfeasance. If the price of the asset being bought has fallen drastically since the locked-in price, an exchanger could intentionally try their best to have the proposal expire so it no longer has a "losing" trade. Regardless, we expect usage of GrandaMento to be infrequent & high visibility, so it's likely that exchanges would be executed quickly when possible.
